### PR TITLE
docs: add `useModalForm` example and documentation

### PR DESCRIPTION
**Pull request by salihozdemir**
Add `useModalForm` example and documentation for `@pankod/refine-react-hook-from` package.

 https://codesandbox.io/embed/github/pankod/refine/tree/use-modal-form-example-and-doc/examples/reactHookForm/useModalForm

### CODE REVIEW

File: examples/reactHookForm/useModalForm/src/components/post/create.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+                </div>
+                <div className="form-group">
+                    <label>Content: </label>
+                    <textarea
+                        {...register("content", { required: true })}
+                        rows={10}
+                    />
+                    {errors.content && <span>This field is required</span>}
+                </div>
+
```
Comment (biskuvit) [inline examples/reactHookForm/useModalForm/src/components/post/create.tsx:37]:
```suggestion
```

File: examples/reactHookForm/useModalForm/src/components/post/edit.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+                </div>
+                <div className="form-group">
+                    <label>Content: </label>
+                    <textarea
+                        {...register("content", { required: true })}
+                        rows={10}
+                    />
+                    {errors.content && <span>This field is required</span>}
+                </div>
+
```
Comment (biskuvit) [inline examples/reactHookForm/useModalForm/src/components/post/edit.tsx:37]:
```suggestion
```

**Review by biskuvit — APPROVED** [review]

File: examples/reactHookForm/useModalForm/src/components/modal.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    children,
+}) => {
+    if (!isOpen) return null;
+    return (
+        <>
+            <div className="overlay" onClick={onClose}></div>
+            <div className="modal">
+                <div className="modal-title">
+                    <button className="close-button" onClick={onClose}>
+                        &times;
```
Comment (ozkalai) [inline examples/reactHookForm/useModalForm/src/components/modal.tsx:20]: I couldn't sure text is true

File: examples/reactHookForm/useModalForm/src/components/post/create.tsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    handleSubmit,
+    modal: { visible, close },
+    saveButtonProps,
+}) => {
+    return (
+        <Modal isOpen={visible} onClose={close}>
+            <form className="form" onSubmit={handleSubmit(onFinish)}>
+                <div className="form-group">
+                    <label>Title: </label>
+                    <input {...register("title", { required: true })} />
```
Comment (ozkalai) [inline examples/reactHookForm/useModalForm/src/components/post/create.tsx:18]:
```suggestion
                    <input {...register("title", { required: "This field is required" })} />
```

File: examples/reactHookForm/useModalForm/src/components/post/create.tsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+                    {errors.title && <span>This field is required</span>}
```
Comment (ozkalai) [inline examples/reactHookForm/useModalForm/src/components/post/create.tsx:19]: If we are going to use required message like above, you can use it from errors object like this: errors.title.message

File: examples/reactHookForm/useModalForm/src/components/post/create.tsx [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+                    <select {...register("status")}>
+                        <option value="published">published</option>
+                        <option value="draft">draft</option>
+                        <option value="rejected">rejected</option>
+                    </select>
+                </div>
+                <div className="form-group">
+                    <label>Content: </label>
+                    <textarea
+                        {...register("content", { required: true })}
```
Comment (ozkalai) [inline examples/reactHookForm/useModalForm/src/components/post/create.tsx:32]:
```suggestion
                        {...register("content", { required: "This field is required" })}
```

File: examples/reactHookForm/useModalForm/src/components/post/create.tsx [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+                        rows={10}
+                    />
+                    {errors.content && <span>This field is required</span>}
```
Comment (ozkalai) [inline examples/reactHookForm/useModalForm/src/components/post/create.tsx:35]:
```suggestion
                    {errors.content && <span>{errors.content.message}</span>}
```

File: examples/reactHookForm/useModalForm/src/components/post/edit.tsx [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+    handleSubmit,
+    modal: { visible, close },
+    saveButtonProps,
+}) => {
+    return (
+        <Modal isOpen={visible} onClose={close}>
+            <form className="form" onSubmit={handleSubmit(onFinish)}>
+                <div className="form-group">
+                    <label>Title: </label>
+                    <input {...register("title", { required: true })} />
```
Comment (ozkalai) [inline examples/reactHookForm/useModalForm/src/components/post/edit.tsx:18]:
```suggestion
                    <input {...register("title", { required: "This field is required" })} />
```

File: examples/reactHookForm/useModalForm/src/components/post/edit.tsx [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+                    {errors.title && <span>This field is required</span>}
```
Comment (ozkalai) [inline examples/reactHookForm/useModalForm/src/components/post/edit.tsx:19]:
```suggestion
                    {errors.title && <span>{errors.title.message}</span>}
```

**Review by aliemir — APPROVED** [review]

**Review by omeraplak — APPROVED** [review]

**Pull request merged**